nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -3,43 +3,52 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-#import hashlib
-#import itertools
 import typing as tp
-import numpy as np
-import sklearn.datasets  # type:ignore
-
 from functools import partial
+import numpy as np
+import sklearn.datasets
+from sklearn.tree import DecisionTreeRegressor
+from sklearn.neural_network import MLPRegressor
 
 from nevergrad.parametrization import parameter as p
-from nevergrad.common import tools
-from nevergrad.common.typetools import ArrayLike
-from sklearn.tree import DecisionTreeRegressor  # type: ignore
-from sklearn.neural_network import MLPRegressor  # type: ignore
-
 from ..base import ExperimentFunction
-from .. import utils
-from .. import corefuncs
-
-# EF = tp.TypeVar("EF", bound="ExperimentFunction")
 
 
+# pylint: disable=too-many-instance-attributes,too-many-locals,too-many-arguments
 class MLTuning(ExperimentFunction):
     """Class for generating ML hyperparameter tuning problems.
+    We propose different possible regressors and different dimensionalities.
+    In each case, Nevergrad will optimize the parameters of a scikit learning.
+
+    Parameters
+    ----------
+    regressor: str
+        type of function we can use for doing the regression. Can be "mlp", "decision_tree", "decision_tree_depth", "any".
+        "any" means that the regressor has one more parameter which is a discrete choice among possibilities.
+    data_dimension: int
+        dimension of the data we generate. None if not an artificial dataset.
+    dataset: str
+        type of dataset; can be diabetes, boston, artificial, artificialcoos, artificialsquare.
+    overfitter: bool
+        if we want the evaluation to be the same as during the optimization run. This means that instead
+        of train/valid/error, we have train/valid/valid. This is for research purpose, when we want to check if an algorithm
+        is particularly good or particularly bad because it fails to minimize the validation loss or because it overfits.
+
     """
 
     # Example of ML problem.
-    def _ml_parametrization(self,
-                            depth: int,  # Parameters for regression trees.
-                            criterion: str, 
-                            min_samples_split: float,
-                            solver: str,  # Parameters for neural nets.
-                            activation: str,
-                            alpha: float,
-                            learning_rate: str,
-                            regressor: str,  # Choice of learner.
-                            noise_free: bool  # Whether we work on the test set (the real cost) on an approximation (CV error on train).
```

Comment:
given the definition, maybe `noise_free` is not the right name, it's not actual noise

Reply:
Yes, you can leave this fix to me for one of these evenings.
Could be "overfitting" or "no_test_error"...

Reply:
`no_cv` or `validation` or `test` (or `train`) would seem more precise, but maybe I dont get it

Reply:
The name "noise_free" was chosen in the sense that the objective function, when it is True, has no randomness, it is deterministic, and evaluation_function == func. In this case,
it is the cross-validation loss.

When it is false, func returns the cross-validation error on the training data, whereas evaluation_function returns the error on separate test data.